### PR TITLE
use bulk_store api when process batches

### PR DESCRIPTION
**Pull request by msgmaxim**

### CODE REVIEW

**Comment by msgmaxim** [comment]
Rebased so that it includes the fix for travis.

File: httpserver/service_node.cpp [hunk 1]
```diff
@@ -363,27 +395,31 @@ void ServiceNode::purge_outdated() {
...
-        return;
-
-    std::vector<message_t> messages = deserialize_messages(*blob);
+    const std::vector<message_t> messages = deserialize_messages(blob);
 
     BOOST_LOG_TRIVIAL(trace) << "saving all: begin";
 
     BOOST_LOG_TRIVIAL(debug) << "got " << messages.size()
-                             << " messages form peers, size: " << blob->size();
+                             << " messages form peers, size: " << blob.size();
```
Comment (BeaudanBrown) [inline httpserver/service_node.cpp]:
```suggestion
                             << " messages from peers, size: " << blob.size();
```

Comment (msgmaxim) [reply httpserver/service_node.cpp]: fixed

File: httpserver/service_node.cpp [hunk 2]
```diff
@@ -363,27 +395,31 @@ void ServiceNode::purge_outdated() {
...
 }
 
-void ServiceNode::process_push_all(std::shared_ptr<std::string> blob) {
-
+void ServiceNode::process_push_batch(const std::string& blob) {
     // Note: we only receive batches on bootstrap (new swarm/new snode)
 
-    /// This should already be checked, but just to be sure
-    if (!blob || *blob == "")
+    if (blob == "")
```
Comment (sachaaaaa) [inline httpserver/service_node.cpp]: `blob.empty()` might be more expressive about the intention?

File: httpserver/service_node.cpp [hunk 3]
```diff
@@ -363,27 +395,31 @@ void ServiceNode::purge_outdated() {
...
-    for (auto& msg : messages) {
-        save_if_new(msg);
-    }
+    std::vector<Item> items;
+    items.reserve(messages.size());
+
+    // Promoting message_t to Item:
+    std::transform(messages.begin(), messages.end(), std::back_inserter(items),
+                   [](const message_t& m) {
+                       return Item{m.hash, m.pub_key,           m.timestamp,
```
[the remark this answers is not in the document]
Comment (msgmaxim) [reply httpserver/service_node.cpp]: Yeah, I don't like it very much, but that was the result of clang-format.

File: httpserver/service_node.cpp (hunk 2 above)
Comment (msgmaxim) [reply httpserver/service_node.cpp]: changed

**Comment by msgmaxim** [comment]
I want to squash/rebase this PR once it has been approved.

File: httpserver/service_node.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (sachaaaaa) [reply httpserver/service_node.cpp]: weird 🤷‍♂️

[the remark this answers is not in the document]
Comment (msgmaxim) [reply httpserver/service_node.cpp]: Could you try running it yourself?

[the remark this answers is not in the document]
Comment (sachaaaaa) [reply httpserver/service_node.cpp]: Just tried, I think it's aligning the fields vertically:
m.hash above m.ttl,
m.pub_key above m.timestamp + m.ttl
m.timestamp above m.nonce

**Pull request merged**